source/Irrlicht/CIrrDeviceSDL.cpp

```diff
@@ -623,6 +632,10 @@ bool CIrrDeviceSDL::run()
 				irrevent.KeyInput.Char = 0;
 				if (mp.SDLKey == SDLK_DELETE || mp.SDLKey == SDLK_RETURN || mp.SDLKey == SDLK_BACKSPACE || irrevent.KeyInput.Control)
 					irrevent.KeyInput.Char = mp.SDLKey;
+				if (mp.SDLKey == SDLK_LSHIFT || mp.SDLKey == SDLK_RSHIFT)
+					IsShiftDown = (SDL_event.type == SDL_KEYDOWN);
+				if (mp.SDLKey == SDLK_LCTRL || mp.SDLKey == SDLK_RCTRL)
+					IsControlDown = (SDL_event.type == SDL_KEYDOWN);

```

Comment:
it really has no better way for this?

Reply:
There's `SDL_GetKeyboardState` to get all keys and `SDL_GetModState` to get modifiers (i.e. shift and control). I guess the latter could be used.
Will do.

Reply:
Done. And rebased.